### PR TITLE
368 invoices with valid payment term discounts are not applied in check run

### PR DESCRIPTION
**Pull request by lauty95**
Resolves https://github.com/agritheory/check_run/issues/368

- Added 2 demo Purchase Invoices and a new Payment Terms Template.
- Added 2 keys to the transaction dictionary for Vue:`discount_amount` and `has_discount` (to show outstanding amount and discount status).
- Added a new field to Check Run Settings: Payment Discount Account.

## Demo

https://github.com/user-attachments/assets/76091a08-ae34-49b2-a494-c81001faaafc

Let me know if this is OK or if I should review any edge cases.

### CODE REVIEW

File: check_run/check_run/doctype/check_run/check_run.py [hunk 1]
```diff
@@ -386,6 +393,17 @@ def create_payment_entries(self, transactions: list[frappe._dict]) -> list[frapp
 				pe.paid_amount = total_amount
```
Comment (HKuz) [inline check_run/check_run/doctype/check_run/check_run.py]: For a Peayment Entry on a PI with a valid Payment Term discount, I believe we need to take the discount amount out of the Paid Amount field. When I duplicate the PI with the valid discount, and just create a Payment Entry in ERPNext (vs using Check Run), that's how they set it up:

Screenshot 2025-07-30 at 7 01 00 PM

**Review by HKuz — CHANGES_REQUESTED** [review]
Nice work, @lauty95 - just had two comments:

- one is in the code, in the Payment Entry, we should reduce the "Paid Amount" field by the discount
- other is for the Check Run - the outstanding amount for any transaction with a valid discount should show the outstanding amount less the discount


Screenshot 2025-07-30 at 7 18 33 PM

File: check_run/check_run/doctype/check_run/check_run.py (hunk 1 above)
Comment (lauty95) [reply check_run/check_run/doctype/check_run/check_run.py]: Changed!
I made a test and now works as a manual Payment Entry. Let me know if the change is ok

**Comment by lauty95** [comment]
@HKuz 

I noticed that the PI "ACC-PINV-2025-00024" is a new PI created by you for testing, correct?

That PI might have an invalid discount. Let me show you what happened when I created a new PI:

https://github.com/user-attachments/assets/33edc40d-d65a-495a-a148-6cf752d81388

Not sure why, but the `Payment Terms Template` was reset.

Also, **Net 14** has no discount to be applied:

Screenshot 2025-08-01 at 11 31 32 AM

Let me show you a quick demo of how I test the discount applied after changing the `Payment Terms Template` again:

https://github.com/user-attachments/assets/f9ed212f-3806-4065-b37f-c53df8f9d3d5

File: check_run/check_run/doctype/check_run/check_run.py (hunk 1 above)
Comment (HKuz) [reply check_run/check_run/doctype/check_run/check_run.py]: @lauty95 works great now, thank you!

**Comment by HKuz** [comment]
@lauty95 - nice work, looks good to me! Yes, the `ACC-PINV-2025-00024` was new while I was testing in the UI. I duplicated the Exceptional Grid one in the test data (because I paid it in a previous Check Run), so it did have the correct discount term. (I think what happened in your demo, is that the AgriTheory supplier in the test data has a default terms of Net 14, so when you changed the Supplier field, it automatically reset the payment terms template with the default). But you're right, the Check Run does show the correct discounted amount - when I ran it, my posting dates were off, so it wasn't showing the discount because it wasn't valid given the dates.

**Review by HKuz — APPROVED** [review]

**Comment by lauty95** [comment]
@HKuz ready the tests.

My aproach was
- first add some `Payment Terms Template` (4 was added in total)
- next, test my function `calculate_payment_term_discount` with some edge cases
- finally, run the `process_check_run` method and tests the final results

Let me know if this looks good to you.
Thanks!

File: check_run/tests/setup.py [hunk 2]
```diff
@@ -261,6 +261,108 @@ def create_payment_terms_templates(settings):
...
+			{
+				"payment_term": pt.name,
+				"invoice_portion": pt.invoice_portion,
+				"due_date_based_on": pt.due_date_based_on,
+				"credit_days": pt.credit_days,
+			},
+		)
+		doc.save()
+
+	if not frappe.db.exists("Payment Terms Template", "3% 10 Net 30"):
```
Comment (HKuz) [inline check_run/tests/setup.py]: Minor point - let's adjust the name of the Payment Term and Template to match the fact it's an amount vs percent - can use something like "$20 10 Net 30"

File: check_run/tests/setup.py [hunk 3]
```diff
@@ -261,6 +261,108 @@ def create_payment_terms_templates(settings):
...
+				"payment_term": pt.name,
+				"invoice_portion": pt.invoice_portion,
+				"due_date_based_on": pt.due_date_based_on,
+				"credit_days": pt.credit_days,
+			},
+		)
+		doc.save()
+
+	# Payment term with "Month(s) after the end of the invoice month" for discount validity
+	if not frappe.db.exists("Payment Terms Template", "3% 10 Net 30 - Month End"):
```
Comment (HKuz) [inline check_run/tests/setup.py]: Let's adjust this name to "3% Net 30" (looks like the discount applies for the full month until the entire invoice is due)

**Review by HKuz — COMMENTED** [review]
Nice work, @lauty95 - just a minor comment on adjusting the Payment Terms/Templates names for clarity. Otherwise, looked great! 👍

**Comment by lauty95** [comment]
@HKuz Ready!

I made a small change to test edge cases more precisely using fixed dates.

**Review by HKuz — APPROVED** [review]
